packages/client/src/clients/guide/client.ts

```diff
@@ -1046,8 +1046,9 @@ export class KnockGuideClient {
 
         return step;
       });
-      // Mutate in place and maintain the same obj ref.
-      guide.steps = steps;
+      // If updated, return the guide as a new object so useStore can trigger.
+      guide = updatedStep ? { ...guide, steps } : guide;
```

Comment:
Didn't realize this but we should update the guide as a new object when updating its step which allows [useStore](https://tanstack.com/store/latest/docs/framework/react/reference/functions/usestore) to trigger which in return allows useGuide to re-trigger.